Fix SpringbootConfigmapBoosterIT on Openshift 3.7

This booster test is showing weird behavior on Openshift 3.7. Ideally, tests are independent of each other and they use their own resources, but when I run this test on my local Openshift cluster; I get this error on my system:
```
Feb 14, 2018 11:58:27 AM io.fabric8.maven.rt.BaseBoosterIT waitTillApplicationPodStarts
INFO: Waiting to application pod .... 
Feb 14, 2018 11:58:27 AM io.fabric8.maven.rt.BaseBoosterIT waitTillApplicationPodStarts
INFO: waitTillApplicationPodStarts() -> Pod : spring-boot-configmap-greeting-18-xpr7r, isReady : true
Feb 14, 2018 11:58:27 AM io.fabric8.maven.rt.BaseBoosterIT waitTillApplicationPodStarts
INFO: OK ✓ ... Pod wait over.
Feb 14, 2018 12:03:37 PM io.fabric8.maven.rt.BaseBoosterIT makeHttpRequest
INFO: [GET] http://spring-boot-configmap-greeting-myproject.127.0.0.1.nip.io/api/greeting [200 OK]
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1,235.662 sec <<< FAILURE! - in io.fabric8.maven.rt.SpringbootConfigmapBoosterIT
deploy_springboot_app_once on deploy_springboot_app_once(io.fabric8.maven.rt.SpringbootConfigmapBoosterIT)(io.fabric8.maven.rt.SpringbootConfigmapBoosterIT)  Time elapsed: 346.242 sec  <<< FAILURE!
java.lang.AssertionError: Actual : Bonjour World from a ConfigMap!, Expected : Hello World from a ConfigMap!
	at io.fabric8.maven.rt.SpringbootConfigmapBoosterIT.assertApplicationEndpoint(SpringbootConfigmapBoosterIT.java:114)
	at io.fabric8.maven.rt.SpringbootConfigmapBoosterIT.assertApplication(SpringbootConfigmapBoosterIT.java:104)
	at io.fabric8.maven.rt.SpringbootConfigmapBoosterIT.deploy_springboot_app_once(SpringbootConfigmapBoosterIT.java:58)


Results :

Failed tests: 
  SpringbootConfigmapBoosterIT.deploy_springboot_app_once:58->assertApplication:104->assertApplicationEndpoint:114 Actual : Bonjour World from a ConfigMap!, Expected : Hello World from a ConfigMap!

Tests run: 2, Failures: 1, Errors: 0, Skipped: 0

```

It is blocking [#1187](https://github.com/fabric8io/fabric8-maven-plugin/issues/1187)

--- a/rt/src/test/java/io/fabric8/maven/rt/BaseBoosterIT.java
+++ b/rt/src/test/java/io/fabric8/maven/rt/BaseBoosterIT.java
@@ -377,7 +377,7 @@ public class BaseBoosterIT {
 
         int nPolls = 0;
         // Keep polling till 5 minutes
-        while (nPolls < 60) {
+        while (nPolls < 120) {
             PodList podList = openShiftClient.pods().withLabel("app", testsuiteRepositoryArtifactId).list();
             for (Pod pod : podList.getItems()) {
                 logger.info("waitTillApplicationPodStarts(" + key + ", " + value + ") -> Pod : "

--- a/rt/src/test/java/io/fabric8/maven/rt/SpringbootConfigmapBoosterIT.java
+++ b/rt/src/test/java/io/fabric8/maven/rt/SpringbootConfigmapBoosterIT.java
@@ -35,7 +35,7 @@ public class SpringbootConfigmapBoosterIT extends BaseBoosterIT {
 
     private final String TESTSUITE_CONFIGMAP_NAME = "app-config";
 
-    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy -DskipTests", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
+    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy -Dfabric8.openshift.trimImageInContainerSpec=true -DskipTests", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
 
     private final String TEST_ENDPOINT = "/api/greeting";
 
@@ -49,9 +49,11 @@ public class SpringbootConfigmapBoosterIT extends BaseBoosterIT {
 
         createViewRoleToServiceAccount();
         createConfigMapResourceForApp(TESTSUITE_CONFIGMAP_NAME, "greeting.message: Hello World from a ConfigMap!");
+        addRedeploymentAnnotations(testRepository, RELATIVE_POM_PATH, "deploymentType", "deployOnce", fmpConfigurationFile);
 
         deploy(testRepository, EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL, EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE);
-        waitAfterDeployment(false);
+        waitTillApplicationPodStarts("deploymentType", "deployOnce");
+        TimeUnit.SECONDS.sleep(20);
         assertApplication(false);
 
         openShiftClient.configMaps().inNamespace(testsuiteNamespace).withName(TESTSUITE_CONFIGMAP_NAME).delete();

--- a/rt/src/test/java/io/fabric8/maven/rt/SpringbootCrudBoosterIT.java
+++ b/rt/src/test/java/io/fabric8/maven/rt/SpringbootCrudBoosterIT.java
@@ -31,7 +31,7 @@ import static io.fabric8.kubernetes.assertions.Assertions.assertThat;
 public class SpringbootCrudBoosterIT extends BaseBoosterIT {
     private final String SPRING_BOOT_CRUD_BOOSTER_GIT = "https://github.com/snowdrop/spring-boot-crud-booster.git";
 
-    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy -DskipTests", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
+    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy -Dfabric8.openshift.trimImageInContainerSpec=true -DskipTests", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
 
     private final String TEST_ENDPOINT = "/api/fruits";
 
@@ -46,8 +46,11 @@ public class SpringbootCrudBoosterIT extends BaseBoosterIT {
         Repository testRepository = setupSampleTestRepository(SPRING_BOOT_CRUD_BOOSTER_GIT, RELATIVE_POM_PATH);
         deployDatabaseUsingCLI();
 
+        addRedeploymentAnnotations(testRepository, RELATIVE_POM_PATH, "deploymentType", "deployOnce", fmpConfigurationFile);
+
         deploy(testRepository, EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL, EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE);
-        waitAfterDeployment(false);
+        waitTillApplicationPodStarts("deploymentType", "deployOnce");
+        TimeUnit.SECONDS.sleep(20);
         assertApplication();
     }
 

--- a/rt/src/test/java/io/fabric8/maven/rt/SpringbootHttpBoosterIT.java
+++ b/rt/src/test/java/io/fabric8/maven/rt/SpringbootHttpBoosterIT.java
@@ -22,6 +22,8 @@ import org.eclipse.jgit.lib.Repository;
 import org.junit.After;
 import org.testng.annotations.Test;
 
+import java.util.concurrent.TimeUnit;
+
 import static io.fabric8.kubernetes.assertions.Assertions.assertThat;
 
 /**
@@ -35,7 +37,7 @@ public class SpringbootHttpBoosterIT extends BaseBoosterIT {
 
     private final String FMP_CONFIGURATION_FILE = "/fmp-plugin-config.xml";
 
-    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
+    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy -Dfabric8.openshift.trimImageInContainerSpec=true", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
 
     private final String RELATIVE_POM_PATH = "/pom.xml";
 
@@ -43,8 +45,11 @@ public class SpringbootHttpBoosterIT extends BaseBoosterIT {
     public void deploy_springboot_app_once() throws Exception {
         Repository testRepository = setupSampleTestRepository(SPRING_BOOT_HTTP_BOOSTER_GIT, RELATIVE_POM_PATH);
 
+        addRedeploymentAnnotations(testRepository, RELATIVE_POM_PATH, "deploymentType", "deployOnce", fmpConfigurationFile);
+
         deploy(testRepository, EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL, EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE);
-        waitUntilDeployment(false);
+        waitTillApplicationPodStarts("deploymentType", "deployOnce");
+        TimeUnit.SECONDS.sleep(20);
         assertApplication();
     }
 

--- a/rt/src/test/java/io/fabric8/maven/rt/VertxConfigmapBoosterIT.java
+++ b/rt/src/test/java/io/fabric8/maven/rt/VertxConfigmapBoosterIT.java
@@ -36,7 +36,7 @@ public class VertxConfigmapBoosterIT extends BaseBoosterIT {
 
     private final String TESTSUITE_CONFIGMAP_NAME = "app-config";
 
-    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy -DskipTests", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
+    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy -Dfabric8.openshift.trimImageInContainerSpec=true -DskipTests", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
 
     private final String TEST_ENDPOINT = "/api/greeting";
 
@@ -52,8 +52,11 @@ public class VertxConfigmapBoosterIT extends BaseBoosterIT {
 
         createViewRoleToServiceAccount();
         createConfigMapResourceForApp(TESTSUITE_CONFIGMAP_NAME);
+        addRedeploymentAnnotations(testRepository, RELATIVE_POM_PATH, "deploymentType", "deployOnce", fmpConfigurationFile);
+
         deploy(testRepository, EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL, EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE);
-        waitAfterDeployment(false);
+        waitTillApplicationPodStarts("deploymentType", "deployOnce");
+        TimeUnit.SECONDS.sleep(20);
         assertDeployment(false);
 
         openShiftClient.configMaps().inNamespace(testsuiteNamespace).withName(TESTSUITE_CONFIGMAP_NAME).delete();

--- a/rt/src/test/java/io/fabric8/maven/rt/VertxHealthchecksBooster.java
+++ b/rt/src/test/java/io/fabric8/maven/rt/VertxHealthchecksBooster.java
@@ -31,7 +31,7 @@ import static io.fabric8.kubernetes.assertions.Assertions.assertThat;
 public class VertxHealthchecksBooster extends BaseBoosterIT {
     private final String SPRING_BOOT_HTTP_BOOSTER_GIT = "https://github.com/openshiftio-vertx-boosters/vertx-health-checks-booster.git";
 
-    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
+    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy -Dfabric8.openshift.trimImageInContainerSpec=true", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
 
     private final String RELATIVE_POM_PATH = "/pom.xml";
 
@@ -41,8 +41,11 @@ public class VertxHealthchecksBooster extends BaseBoosterIT {
     public void deploy_vertx_app_once() throws Exception {
         Repository testRepository = setupSampleTestRepository(SPRING_BOOT_HTTP_BOOSTER_GIT, RELATIVE_POM_PATH);
 
+        addRedeploymentAnnotations(testRepository, RELATIVE_POM_PATH, "deploymentType", "deployOnce", fmpConfigurationFile);
+
         deploy(testRepository, EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL, EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE);
-        waitAfterDeployment(false);
+        waitTillApplicationPodStarts("deploymentType", "deployOnce");
+        TimeUnit.SECONDS.sleep(20);
         assertDeployment();
     }
 

--- a/rt/src/test/java/io/fabric8/maven/rt/VertxHttpBoosterIT.java
+++ b/rt/src/test/java/io/fabric8/maven/rt/VertxHttpBoosterIT.java
@@ -33,7 +33,7 @@ import static io.fabric8.kubernetes.assertions.Assertions.assertThat;
 public class VertxHttpBoosterIT extends BaseBoosterIT {
     private final String SPRING_BOOT_HTTP_BOOSTER_GIT = "https://github.com/openshiftio-vertx-boosters/vertx-http-booster.git";
 
-    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
+    private final String EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL = "fabric8:deploy -Dfabric8.openshift.trimImageInContainerSpec=true", EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE = "openshift";
 
     private final String RELATIVE_POM_PATH = "/pom.xml";
 
@@ -45,8 +45,11 @@ public class VertxHttpBoosterIT extends BaseBoosterIT {
     public void deploy_vertx_app_once() throws Exception {
         Repository testRepository = setupSampleTestRepository(SPRING_BOOT_HTTP_BOOSTER_GIT, RELATIVE_POM_PATH);
 
+        addRedeploymentAnnotations(testRepository, RELATIVE_POM_PATH, "deploymentType", "deployOnce", fmpConfigurationFile);
+
         deploy(testRepository, EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL, EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE);
-        waitTillApplicationPodStarts();
+        waitTillApplicationPodStarts("deploymentType", "deployOnce");
+        TimeUnit.SECONDS.sleep(20);
         assertDeployment();
     }
 

